### PR TITLE
feat: verification in progress

### PR DESCRIPTION
**Pull request by dgacevic**
https://bayes.atlassian.net/browse/RW-1026
also solves https://bayes.atlassian.net/browse/RW-1045

### CODE REVIEW

File: brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressInteractor.swift [hunk 1]
```diff
@@ -60,6 +60,29 @@ class KYCAddressInteractor: NSObject, Interactor, KYCAddressViewActions {
...
+        
+        KYCSubmitInfoWorker().execute(requestData: data) { [weak self] result in
+            switch result {
+            case .success:
+                self?.startExternalKYC(viewAction: .init())
+                
+            case .failure(let error):
+                self?.presenter?.presentError(actionResponse: .init(error: error))
+            }
+
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressInteractor.swift:82]: remove line

File: brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressInteractor.swift [hunk 2]
```diff
@@ -60,6 +60,29 @@ class KYCAddressInteractor: NSObject, Interactor, KYCAddressViewActions {
...
+    func submitInfo(viewAction: KYCAddressModels.Submit.ViewAction) {
+        let data = KYCUserInfoRequestData(firstName: dataStore?.firstName ?? "",
+                                          lastName: dataStore?.lastName ?? "",
+                                          dateOfBirth: dataStore?.birthDateString ?? "",
+                                          address: dataStore?.address ?? "",
+                                          city: dataStore?.city ?? "",
+                                          zip: dataStore?.postalCode ?? "",
+                                          country: dataStore?.country ?? "",
+                                          state: dataStore?.state,
+                                          nologSSN: nil)
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCAddressInteractor.swift:72]: please add todo

File: brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCSubmitInfoWorker.swift [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        
+        if let state = state {
+            params["state"] = state
+        }
+        
+        if let nologSSN = nologSSN {
+            params["nolog_ssn"] = nologSSN
+        }
+        
+        return params
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCSubmitInfoWorker.swift:43]: image

wouldn't this approach work?

File: brd-ios/breadwallet/src/Redesign/Scenes/REUSABLE/CheckList/CheckListInteractor.swift [hunk 4]
```diff
@@ -15,6 +15,41 @@ class CheckListInteractor: NSObject, Interactor, CheckListViewActions {
...
+                    self?.presenter?.presentVerificationProgress(actionResponse: .init(status: .failure(.documentVerification)))
+                    
+                case .levelTwo(.resubmit):
+                    self?.presenter?.presentVerificationProgress(actionResponse: .init(status: .failure(.documentVerificationRetry)))
+                    
+                    
+                default:
+                    // waiting for verification, check every 30 seconds
+                    DispatchQueue.main.asyncAfter(deadline: .now() + 30, execute: {
+                        print("checking verification status again")
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/REUSABLE/CheckList/CheckListInteractor.swift:38]: please, only use debugPrint or remove it. It is not useful here. And capitalize first letters please :)

File: brd-ios/breadwallet/src/Redesign/Views/VerificationView.swift [hunk 5]
```diff
@@ -17,6 +17,8 @@ enum Kyc2: String, Equatable {
     case levelTwo = "KYC2"
     case resubmit = "KYC2_RESUBMISSION_REQUESTED"
     case declined = "KYC2_DECLINED"
+    case KYCwithSSN = "KYC_WITH_SSN"
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Views/VerificationView.swift:20]: kycWithSsn

File: brd-ios/breadwallet/src/Redesign/Views/VerificationView.swift [hunk 6]
```diff
@@ -17,6 +17,8 @@ enum Kyc2: String, Equatable {
...
+    case KYCwithoutSSN = "KYC_WITHOUT_SSN"
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Views/VerificationView.swift:21]: kycWithoutSsn

File: brd-ios/breadwallet/src/Redesign/Scenes/REUSABLE/CheckList/CheckListInteractor.swift (hunk 4 above)
Comment (dgacevic) [reply brd-ios/breadwallet/src/Redesign/Scenes/REUSABLE/CheckList/CheckListInteractor.swift:38]: Don't actually need this, removed

File: brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCSubmitInfoWorker.swift [hunk 7]
```diff
@@ -40,7 +40,7 @@ struct KYCUserInfoRequestData: RequestModelData {
             params["nolog_ssn"] = nologSSN
         }
         
-        return params
+        return params.compactMapValues { $0 }
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCSubmitInfoWorker.swift:43]: are the if checks above necessary after compactMapValues?

Comment (dgacevic) [reply brd-ios/breadwallet/src/Redesign/Scenes/KYC/KYCAddress/KYCSubmitInfoWorker.swift:43]: You are right, will move them out of if

**Review by kenanmamedoff — APPROVED** [review]

**Pull request merged**